src/soho/colorpicker/soho-colorpicker.component.ts

```diff
@@ -88,6 +96,70 @@ export class SohoColorPickerComponent extends BaseControlValueAccessor<any> impl
     }
   }
 
+  /**
+   * Enables or disables forceing upper case hex.
+   */
+  @Input() set uppercase(value: boolean) {
+    this.isUppercase = value;
+    this.options.uppercase = value;
+
+    if (this.colorpicker) {
+      this.colorpicker.settings.uppercase = value;
+    }
+
+    if (this.colorpicker) {
+      this.colorpicker.updated(this.options);
```

Comment:
would like to see a `callUpdatedAfterCheck: boolean` flag then call the updated() function in a runOutsideAngular in ngAfterViewChecked.

Example
In the @Inputs just set the flag.
```
  if (this.colorpicker) {
      this.callUpdatedAfterCheck = true;
  }
```

In ngAfterViewChecked check the flag and then call updated and reset the flag
```
ngAfterViewChecked() {
  if (this.callUpdatedAfterCheck) {
    this.ngZone.runOutsideAngular(() => this.colorpicker.updated(this.options));
    this.callUpdatedAfterCheck = false;
  }
}
```

Maybe not super important right now but it's a good way to ensure that updated() is not called for each input setting but instead at the end of a tick() after all inputs have been set.

Reply:
I can make the change. Is there a reference component done with all ngZone updates and the change to only trigger on ngAfterViewChecked ?

Reply:
I would take a look at dropdown, as this has both the run updated on check boolean, and the NgZone code.